client/src/featureform/register.py

```diff
@@ -2572,6 +2572,50 @@ def apply(self):
         else:
             state().create_all(self._stub)
 
+    def wait(self, resource_type, name, variant=None, timeout=None):
+        # gets a resource and waits until status is set to ready
+        resource_functions = {
+            "user": self.get_user,
+            "model": self.get_model,
+            "entity": self.get_entity,
+            "provider": self.get_provider
+        }
+
+        resource_variant_functions = {
+            "feature": self.get_feature,
```

Comment:
Isnt this essentially the resource object getting itself? Why not use `self`?

Reply:
It is. But the ResourceClient has the stub for connection and each function (listed here), and each resource has its data, so this function is passed to each resource, and they provide their data (name, variant, type)

Reply:
But cant the resource pass itself in? Why not just have `wait(self, obj, timeout)` and below do `self.wait_function(self, timeout)` ?